pkg/juicefs/mount/pod_mount.go

```diff
@@ -173,64 +183,60 @@ func (p *PodMount) waitUntilMount(jfsSetting *jfsConfig.JfsSetting, volumeId, ta
 	}
 
 	key := util.GetReferenceKey(target)
-	needCreate := false
-	for i := 0; i < 30; i++ {
+	for i := 0; i < 120; i++ {
 		// wait for old pod deleted
 		if oldPod, err := p.K8sClient.GetPod(podName, jfsConfig.Namespace); err == nil && oldPod.DeletionTimestamp != nil {
-			klog.Infof("waitUtilMount: wait for old mount pod deleted.")
+			klog.V(6).Infof("createOrAddRef: wait for old mount pod deleted.")
 			time.Sleep(time.Millisecond * 500)
 			continue
 		} else if err != nil {
 			if k8serrors.IsNotFound(err) {
-				needCreate = true
-				break
+				// pod not exist, create
+				klog.V(5).Infof("createOrAddRef: Need to create pod %s.", podName)
+				newPod := NewMountPod(podName, cmd, mountPath, podResource, config, env, labels, anno, serviceAccount)
+				if newPod.Annotations == nil {
+					newPod.Annotations = make(map[string]string)
+				}
+				newPod.Annotations[key] = target
+				_, err = p.K8sClient.CreatePod(newPod)
+				if err != nil {
+					klog.Errorf("createOrAddRef: Create pod %s err and retry: %v", podName, err)
+				}
+				return nil
 			}
-			klog.Errorf("get pod err:%v", err)
+			// unexpect error
+			klog.Errorf("createOrAddRef: Get pod %s err and retry: %v", podName, err)
 			return err
 		}
-		break
-	}
-
-	if needCreate {
-		klog.V(5).Infof("waitUtilMount: Need to create pod %s.", podName)
-		newPod := NewMountPod(podName, cmd, mountPath, podResource, config, env, labels, anno, serviceAccount)
-		if newPod.Annotations == nil {
-			newPod.Annotations = make(map[string]string)
-		}
-		newPod.Annotations[key] = target
-		if _, e := p.K8sClient.CreatePod(newPod); e != nil {
-			return e
-		}
+		// pod exist, add refs
+		return p.AddRefOfMount(target, podName)
 	}
+	return status.Errorf(codes.Internal, "Mount %v failed: mount pod %s has been deleting for 1 min", volumeId, podName)
+}
 
+func (p *PodMount) waitUtilPodReady(volumeId string) error {
+	podName := GeneratePodNameByVolumeId(volumeId)
 	// Wait until the mount pod is ready
-	for i := 0; i < 30; i++ {
+	for i := 0; i < 60; i++ {
 		pod, err := p.K8sClient.GetPod(podName, jfsConfig.Namespace)
 		if err != nil {
-			return status.Errorf(codes.Internal, "waitUtilMount: Get pod %v failed: %v", volumeId, err)
+			return status.Errorf(codes.Internal, "waitUtilPodReady: Get pod %v failed: %v", volumeId, err)
```

Comment:
The pod could be deleted and recovered, so we should retry when error is returned

Reply:
Call apiServer error is unexpected error, we should return it and wait for kubelet retry